operator/v1/types_network.go

```diff
@@ -561,7 +563,35 @@ type IPsecConfig struct {
 	// this is left to the user to configure.
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
+	// +unionDiscriminator
 	Mode IPsecMode `json:"mode,omitempty"`
+
+	// full defines configuration parameters for the IPsec `Full` mode.
+	// This is permitted only when mode is configured with `Full`,
+	// and forbidden otherwise.
+	// +unionMember,optional
+	// +optional
+	Full *IPsecFullModeConfig `json:"full,omitempty"`
+}
+
+type Encapsulation string
+
+const (
+	// ForceEncapsulation forces UDP encapsulation regardless of whether NAT is detected.
+	ForceEncapsulation = "Force"
+	// DisableEncapsulation disables UDP encapsulation even if NAT is present.
+	DisableEncapsulation = "Disable"
+	// AutoEncapsulation determines use of UDP encapsulation based on the detection of NAT.
+	AutoEncapsulation = "Auto"
+)
+
+type IPsecFullModeConfig struct {
+	// encapsulation option to configure libreswan on how inter-pod traffic across nodes
+	// are encapsulated to handle NAT traversal. When configured it uses UDP port 4500
+	// for the encapsulation.
```

Comment:
I think the purpose of this field is documented well in this GoDoc, but I would like to see more information outlined on:
- Limitations of this field. I would recommend explicitly calling out the allowed values in the GoDoc and describing how each setting influences the behavior
- As of writing, this field is optional and doesn't look to have a default - what does not setting this field mean for encapsulation?

Reply:
As Bryce says, we need to add godoc explaining the options, this typically takes the form of

```
// Valid values are Force, Disable, Auto and omitted.
// Force means ... 
// Disable means ...
// Auto means ...
// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
// The current default is ...
```

Also, given this is optional/omitempty you don't need `""` as a valid value in the enum

Reply:
sure, updated godoc with above info, hope it's fine now.